src/patterns/menu/Menu.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import Text from '../../components/text';
+import Shape from '../../foundations/shape';
+import Grid from '../../utilities/layout';
+import Popover from '../../utilities/popover';
+import Spacing from '../../utilities/spacing';
+
+import type PMenu from './Menu.props';
+
+const Menu = ({
+    trigger,
+    renderTrigger = true,
+    title,
+    width,
+    container,
+    groups,
+    hasSubmenu,
+    isVisible,
+    ...rest
+}: PMenu): JSX.Element => {
+    const divider = <Shape height="1px" width={'auto'} backgroundColor={'#e0e0e0'}></Shape>;
+    return (
+        <>
+            {renderTrigger && trigger.element}
+            <Popover
+                isVisible={isVisible}
+                anchorReference={trigger.ref}
+                onClickAway={(): void => {}}
+                {...rest}
+            >
+                <Shape element={container} width={width} elevation={hasSubmenu ? 5 : 4} radius={8}>
+                    <Grid flex={1} justifyItems={'left'} alignItems={'center'}>
```

Comment:
you can combine these components within the Grid component (see here: https://github.com/mattermost/compass-components/blob/f6c02829cf5a85dfe817d816ee63cb5407572c91/src/utilities/grid/Grid.tsx#L14-L22)